src/snowflake/snowpark/_internal/utils.py

```diff
@@ -839,10 +847,29 @@ def check_is_pandas_dataframe_in_to_pandas(result: Any) -> None:
 def get_copy_into_table_options(
     options: Dict[str, Any]
 ) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+    """Method that extracts options for COPY INTO TABLE command into file
+    format type options and copy options.
+    """
+    file_format_type_options = options.get("FORMAT_TYPE_OPTIONS", {})
+    copy_options = options.get("COPY_OPTIONS", {})
+    for k, v in options.items():
+        if k in COPY_INTO_TABLE_COPY_OPTIONS:
+            copy_options[k] = v
+        elif k not in NON_FORMAT_TYPE_OPTIONS:
+            file_format_type_options[k] = v
+    return file_format_type_options, copy_options
```

Comment:
These two function are almost entirely the same. It might be cleaner to make something like this in order to share logic:
```python
def _get_options(
    options: Dict[str, Any],
    allowed_options: Dict[str, Any],
) -> Tuple[Dict[str, Any], Dict[str, Any]]:
    """Method that extracts options for COPY INTO LOCATION command into file
    format type options and copy options.
    """
    file_format_type_options = options.get("FORMAT_TYPE_OPTIONS", {})
    copy_options = options.get("COPY_OPTIONS", {})
    for k, v in options.items():
        if k in COPY_OPTIONS:
        if k in allowed_options:
            copy_options[k] = v
        elif k not in NON_FORMAT_TYPE_OPTIONS:
            file_format_type_options[k] = v
    return file_format_type_options, copy_options

get_copy_into_table_options = partial(_get_options, allowed_options=COPY_INTO_TABLE_COPY_OPTIONS)
get_copy_into_location_options = partial(_get_options, allowed_options= COPY_INTO_LOCATION_COPY_OPTIONS)
```

Reply:
that's a great suggestion. Updated